### PR TITLE
redesign gallery chart component

### LINKED ISSUE #4235: redesign: Chart Component and Tabs section
requirements: https://github.com/kodadot/nft-gallery/issues/4111#chart-section

### PR DESCRIPTION
**Pull request by Jarsen136**
**Thank you for your contribution** to the [KodaDot NFT gallery](https://kodadot.xyz).

👇 \_ Let's make a quick check before the contribution.

## PR Type

- [ ] Bugfix
- [x] Feature
- [x] Refactoring

## Context

- [x] Closes #4235
- [ ] Requires deployment <>

#### Before submitting pull request, please make sure:

- [x] My contribution builds **clean without any errors or warnings**
- [x] I've merged recent default branch -- **main** and I've no conflicts
- [x] I've tried to respect high code quality standards
- [x] I've didn't break any original functionality
- [x] I've posted a screenshot of demonstrated change in this PR

#### Optional

- [ ] I've tested it at </bsx/collection>
- [ ] I've tested PR on mobile
- [ ] I've written unit tests 🧪
- [ ] I've found edge cases

#### Had issue bounty label?

- [x] Fill up your KSM address: [Payout](https://beta.kodadot.xyz/transfer/?target=Caiv9TbPz68q5dC8EcHu5xKYPRnremimGzqmEejDFNpWWLG)

#### Community participation

- [ ] [Are you at KodaDot Discord?](https://discord.gg/35hzy2dXXh)

## Screenshot 📸

- [x] My fix has changed **something** on UI; a screenshot is best to understand changes for others.
https://deploy-preview-4355--koda-nuxt.netlify.app/rmrk/gallery/11823638-900D19DC7D3C444E4C-KSMBOT-KUSAMA_BOT_063-0000000000000066?redesign=true

image

image

### CODE REVIEW

**Comment by Jarsen136** [comment]
> is there a particular item we are testing chart data?

https://deploy-preview-4355--koda-nuxt.netlify.app/rmrk/gallery/11823638-900D19DC7D3C444E4C-KSMBOT-KUSAMA_BOT_063-0000000000000066?redesign=true

randomly grab a item from here
image

**Comment by Jarsen136** [comment]
Missing: https://github.com/kodadot/nft-gallery/issues/4235#issuecomment-1301624209
>create a chart wrapper under histoire (put component under libs/ui / @kodadot1/brick package) and using this package https://vue-chartjs.org/guide/

File: libs/ui/src/components/Chart/PriceChart.vue [hunk 1]
```diff
@@ -0,0 +1,265 @@
...
+    value: 0,
+    label: 'All',
+  },
+  {
+    value: 14,
+    label: '14 days',
+  },
+  {
+    value: 30,
+    label: '30 days',
```
Comment (prachi00) [inline libs/ui/src/components/Chart/PriceChart.vue:47]: translations for these?

File: libs/ui/src/components/Chart/PriceChart.vue [hunk 2]
```diff
@@ -0,0 +1,265 @@
...
+
+const props = defineProps<{
+  priceChartData?: [Date, number][][]
+  isDarkMode?: boolean
+  unit?: string
+}>()
+let Chart: ChartJS<'line', any, unknown>
+
+onMounted(() => {
+  console.log('jarsen unit ', props, props.unit)
```
Comment (prachi00) [inline libs/ui/src/components/Chart/PriceChart.vue:69]: remove this?

File: libs/ui/src/components/Chart/PriceChart.vue (hunk 1 above)
Comment (Jarsen136) [reply libs/ui/src/components/Chart/PriceChart.vue]: done

File: libs/ui/src/components/Chart/PriceChart.vue (hunk 2 above)
Comment (Jarsen136) [reply libs/ui/src/components/Chart/PriceChart.vue]: done

**Review by prachi00 — APPROVED** [review]

**Review by roiLeo — COMMENTED** [review]
I find that a lot of logic has been written in histoire component and I don't think this is the best approach if we want to reuse this component. I would only display canvas in component.

Is Chart.js usage better compared with vue-chartjs wrapper?

File: libs/ui/package.json [hunk 3]
```diff
@@ -20,6 +20,7 @@
   },
   "dependencies": {
     "@google/model-viewer": "^1.12.1",
+    "buefy": "^0.9.22",
```
[the remark this answers is not in the document]
Comment (Jarsen136) [reply libs/ui/package.json]: You're right. I removed it

File: libs/ui/src/components/Chart/PriceChart.vue [hunk 4]
```diff
@@ -0,0 +1,266 @@
...
+
+<script lang="ts" setup>
+import { computed, onBeforeUnmount, onMounted, ref, watch } from 'vue'
+import ChartJS from 'chart.js/auto'
+import 'chartjs-adapter-date-fns'
+import zoomPlugin from 'chartjs-plugin-zoom'
+import { getChartData } from '@/utils/chart'
+ChartJS.register(zoomPlugin)
+
+const { $i18n } = useNuxtApp()
```
[the remark this answers is not in the document]
Comment (Jarsen136) [reply libs/ui/src/components/Chart/PriceChart.vue]: Yes, it would work

**Comment by Jarsen136** [comment]
> I find that a lot of logic has been written in histoire component and I don't think this is the best approach if we want to reuse this component. I would only display canvas in component.
> 
> Is Chart.js usage better compared with vue-chartjs wrapper?

Make sense. I would create another issue to migrate Chart.js to vue-chartjs and put it in histoire component.

File: components/chart/PriceChart.vue [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+    :unit="unit"
+    :price-chart-data="priceChartData" />
+</template>
+
+<script lang="ts" setup>
+import { getChartData } from '@/utils/chart'
+import useChain from '@/composables/useChain'
+import { PriceChart } from '@kodadot1/brick'
+
+const { $colorMode } = useNuxtApp()
```
[the remark this answers is not in the document]
Comment (Jarsen136) [reply components/chart/PriceChart.vue]: It seems does not work : )

**Comment by yangwao** [comment]
wfm!

i'll wait for @exezbcz to give it `✅ visual ok`

**Comment by exezbcz** [comment]
### Visual changes
image
design: 
image

- the lines should have different color. 
main lines, x;y are black, the others are grey. (shade in figma) 
- Can you make it so that there aren't as many dates?

hover on point.
- point should change -> background to background color(in this case white), stroke - default color of point. 
- like this: 
image

hover popup
- popup has this design: 
image
in gallery item handoff there was a bit different design, so i simplified it and we will only show the price and date

button for time frame
- text should be 12px
- padding same as secondary buttons
- current:
image
- design: 
image

**Comment by yangwao** [comment]
I've topped up bounty @Jarsen136, as this is a bit beyond the scope of the original issue! :)

**Comment by Jarsen136** [comment]
Ok, I will try to build this part

File: components/chart/PriceChart.vue (hunk 5 above)
[the remark this answers is not in the document]
Comment (roiLeo) [reply components/chart/PriceChart.vue]: > It seems does not work : )

then wrong `@nuxtjs/color-mode` configuration

File: libs/ui/src/components/Chart/PriceChart.vue (hunk 4 above)
[the remark this answers is not in the document]
Comment (roiLeo) [reply libs/ui/src/components/Chart/PriceChart.vue]: What if in a near future we move `libs/ui` package outside this project, we won't be able to `useNuxtApp`. 
IMO this should stay independant.

[the remark this answers is not in the document]
Comment (roiLeo) [reply libs/ui/src/components/Chart/PriceChart.vue]: nevermind you remove histoire component

**Review by roiLeo — APPROVED** [review]
✅ code lgtm

### refacto:
- with #4430 design for `PriceChart` component 
- create histoire component

**Comment by yangwao** [comment]
@exezbcz give it one more visual check please 🥺

**Comment by exezbcz** [comment]
yup
### dropdown
image
- Please style the menus to look like profile dropdowns.

### tool tip

popup has this design:
image

### dark mode lines
image

x lines should be grey

**Comment by yangwao** [comment]
> nevermind, good job

we can still address missing pieces 
- https://github.com/kodadot/nft-gallery/issues/4403

**Comment by yangwao** [comment]
pay 200 usd

let's resolve merge conflicts and feel free to merge this PR; we are close to finishing the gallery item! :)

**Comment by yangwao** [comment]
😍 Perfect, I’ve sent the payout 
💵 $200 @ 30.09 USD/KSM ~ 6.647 $KSM
🧗 [Caiv9TbPz68q5dC8EcHu5xKYPRnremimGzqmEejDFNpWWLG](https://kusama.subscan.io/account/Caiv9TbPz68q5dC8EcHu5xKYPRnremimGzqmEejDFNpWWLG)
🔗 [0xb987f9d2a6342bad8a34f8d6c6abdeef79fb07f83ae3b7f5824b10d3da7ab82d](https://kusama.subscan.io/extrinsic/0xb987f9d2a6342bad8a34f8d6c6abdeef79fb07f83ae3b7f5824b10d3da7ab82d)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)

**Pull request merged**